src/SearchTrips/Trip.tsx

```diff
@@ -0,0 +1,156 @@
+    .map((leg) => ({
+      icon: getTransportIcon(leg.mode, leg.transportSubmode),
+      text: `${leg.line?.publicCode || ""}`,
+    }));
+  accessories.push({
+    tag: {
+      value: formatTimeDifferenceAsClock(trip.expectedStartTime, trip.expectedEndTime),
+      color: Color.Blue,
+    },
+  });
+  return accessories;
+};
+
+type Props = {
+  origin: Feature;
+  destination: Feature;
+  strategy: "arrive-by" | "leave-at";
+};
+export default function Trip({ origin, destination, strategy }: Props) {
+  const [trips, setTrips] = useState<TripPattern[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [pageCursor, setPageCursor] = useState("");
+  const [isDetailVisible, setDetailVisible] = useState(false);
+  const [dateTime] = useState(new Date());
+
+  const getTrips = useCallback(() => {
+    setIsLoading(true);
+    fetchTrip({
+      originId: origin.properties.id,
+      destinationId: destination.properties.id,
+      arriveBy: strategy === "arrive-by",
+      dateTime,
+      pageCursor,
+    })
+      .then((data) => {
+        const newTrips = trips?.concat(data.trip.tripPatterns);
+        setTrips(newTrips);
+        setPageCursor(data.trip.nextPageCursor);
+      })
+      .finally(() => setIsLoading(false));
+  }, [pageCursor]);
+
+  useEffect(() => {
+    getTrips();
+  }, []);
+
+  return (
+    <List
+      isLoading={isLoading}
+      isShowingDetail={isDetailVisible}
+      searchBarPlaceholder={`From ${origin.properties.name} to ${destination.properties.name}...`}
+    >
+      {trips?.map((trip, idx) => (
+        <List.Item
+          detail={<List.Item.Detail metadata={<TripDetails trip={trip} />}></List.Item.Detail>}
+          actions={
+            <ActionPanel>
+              <Action
+                title="Toggle Details"
+                onAction={() => {
+                  setDetailVisible(!isDetailVisible);
+                }}
+              />
+              <Action
+                title="Load More"
+                onAction={() => {
+                  getTrips();
+                  /**
+                   * Perhaps we could remove this in favor of
+                   * `pagination` returned from using the built-in useFetch/usePromise
+                   * [more here](https://developers.raycast.com/utilities/react-hooks/usefetch#pagination)
+                   *
+                   */
+                }}
+              />
+            </ActionPanel>
+          }
+          title={`${formatAsClock(trip.expectedStartTime)} - ${formatAsClock(trip.expectedEndTime)}`}
+          // TODO: subtitle => "Little warning triangle if service disruptions are present"
+          accessories={buildAccessories(trip, isDetailVisible)}
+          key={idx}
+        />
+      ))}
+    </List>
+  );
+}
+
+const TripDetails = ({ trip }: { trip: TripPattern }) => {
+  const getTitleText = (leg: Leg, isArriveText?: boolean) => {
```

Comment:
I'm more in favor of spitting this in two functions, e.g `getLegTitle` and `getDestinationTitle`. That might be slightly more readable.